### ISSUE TITLE
calicoctl leaks `/tmp/_MEI*` directories on start

### ISSUE THREAD

**Issue by xh3b4sd** [projectcalico/calico-containers#1178]
This issue is similar to https://github.com/projectcalico/calico-containers/issues/942. It is only that it does not effect the `calico-cni` plugin, but `calicoctl` itself. Further the leaked directories are created when running `calicoctl node --ip=${DEFAULT_IPV4} --detach=false --node-image=giantswarm/node:v0.22.0`. Note that the image giantswarm/node:v0.22.0 is only re-tagged from the original release. So on each start a directory `/tmp/_MEI*` is created and fills the disk. I had some VMs running over night which stopped doing their jobs at some point because there was no more space left on the device. FYI I ran some resilience tests where I restarted calico all 5 minutes. This restarting behaviour is not intended be used in an production environment but considering some restart of some tool causes the disk to be filled is rather bad. I have an work around for my systemd unit files where I define this statement.

```
ExecStopPost=/bin/bash -c "find /tmp/ -name '_MEI*' | xargs -I {} rm -rf {}"
```

Not sure if this is a good idea and for what these tmp directories are really used for. Maybe some folks can shed some light on this story. Cheers.

**Comment by matthewdupre** [comment]
These are created by pyinstaller: the way we're now distributing Felix for calico-node.  See https://pythonhosted.org/PyInstaller/operating-mode.html#how-the-one-file-program-works

It should always be safe to delete them as long as calicoctl isn't running.  I also expect the problem will go away completely in the next couple of months when we replace the current calicoctl with one written in go (which does a much better job being a static binary than pyinstaller).

**Comment by fasaxc** [comment]
@matthewdupre Felix is packaged as a single-dir PyInstaller bundle to avoid this problem.  I think the issue is that calico/node now uses PyInstaller to package some of the other python apps that are in calico/node as well as calicoctl itself.

**Comment by xh3b4sd** [comment]
Hello there. I wanted to check back and see if there are any new information on this issue. Cheers.

**Comment by caseydavenport** [comment]
Fixed in the v1.0.0-beta, soon to be released as v1.0.0.
